### PR TITLE
Unify 1.8 and 1.9

### PR DESCRIPTION
**Pull request by tigerw**
- Deleted ridiculous amount of duplicated code

In order to do this I ran on `master`:

`git diff --no-index -- src\Protocol\Protocol_1_8.cpp src\Protocol\Protocol_1_9.cpp`
`git diff --no-index -- src\Protocol\Protocol_1_8.cpp src\Protocol\Protocol_1_9.cpp`

and deleted the functions that were identical.

This is to prepare for xoft's request to add a SendEntity function, which is expected to work almost identically in every protocol version.

### CODE REVIEW

**Comment by madmaxoft** [comment]
Have you tested both client versions after the change? I think there was a deeper reason for having the protocols separate, maybe it was due to packet compression or something like that.

File: src/Protocol/Protocol_1_8.h [hunk 1]
```diff
@@ -191,14 +191,14 @@ class cProtocol_1_8_0:
...
 	void HandlePacketPlayerLook             (cByteBuffer & a_ByteBuffer);
 	void HandlePacketPlayerPos              (cByteBuffer & a_ByteBuffer);
 	void HandlePacketPlayerPosLook          (cByteBuffer & a_ByteBuffer);
-	void HandlePacketPluginMessage          (cByteBuffer & a_ByteBuffer);
-	void HandlePacketResourcePackStatus     (cByteBuffer & a_ByteBuffer);
+	virtual void HandlePacketPluginMessage          (cByteBuffer & a_ByteBuffer);
+	virtual void HandlePacketResourcePackStatus     (cByteBuffer & a_ByteBuffer);
 	void HandlePacketSlotSelect             (cByteBuffer & a_ByteBuffer);
 	void HandlePacketSpectate               (cByteBuffer & a_ByteBuffer);
 	void HandlePacketSteerVehicle           (cByteBuffer & a_ByteBuffer);
```
Comment (madmaxoft) [inline src/Protocol/Protocol_1_8.h:204]: This needs fixing. Make all the functions `virtual`.

**Comment by tigerw** [comment]
I tested loading the world, sitting on a pressure plate, hitting a minecart to get it to shake, acquiring splash potions and spawn eggs, throwing splash potions, firing off arrows and going through a portal, on versions 1.8.9, 1.9, 1.12. By no means comprehensive of course.

**Comment by tigerw** [comment]
I'm gonna do it.

**Comment by tigerw** [comment]
Fingers on the revert button if something goes wrong.

**Pull request merged**